sdk/go/x/auto/stack.go

```diff
@@ -0,0 +1,772 @@
+		return res, newAutoError(errors.Wrap(err, "failed to destroy stack"), stdout, stderr, code)
+	}
+
+	history, err := s.History(ctx)
+	if err != nil {
+		return res, errors.Wrap(err, "failed to destroy stack")
+	}
+
+	var summary UpdateSummary
+	if len(history) > 0 {
+		summary = history[0]
+	}
+
+	res = DestroyResult{
+		Summary: summary,
+		StdOut:  stdout,
+		StdErr:  stderr,
+	}
+
+	return res, nil
+}
+
+// Outputs get the current set of Stack outputs from the last Stack.Up().
+func (s *Stack) Outputs(ctx context.Context) (OutputMap, error) {
+	err := s.Workspace().SelectStack(ctx, s.Name())
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get stack outputs")
+	}
+
+	// standard outputs
+	outStdout, outStderr, code, err := s.runPulumiCmdSync(ctx, "stack", "output", "--json")
+	if err != nil {
+		return nil, newAutoError(errors.Wrap(err, "could not get outputs"), outStdout, outStderr, code)
+	}
+
+	// secret outputs
+	secretStdout, secretStderr, code, err := s.runPulumiCmdSync(ctx, "stack", "output", "--json", "--show-secrets")
+	if err != nil {
+		return nil, newAutoError(errors.Wrap(err, "could not get secret outputs"), outStdout, outStderr, code)
+	}
+
+	var outputs map[string]interface{}
+	var secrets map[string]interface{}
+
+	if err = json.Unmarshal([]byte(outStdout), &outputs); err != nil {
+		return nil, errors.Wrapf(err, "error unmarshalling outputs: %s", secretStderr)
+	}
+
+	if err = json.Unmarshal([]byte(secretStdout), &secrets); err != nil {
+		return nil, errors.Wrapf(err, "error unmarshalling secret outputs: %s", secretStderr)
+	}
+
+	res := make(OutputMap)
+	for k, v := range secrets {
+		isSecret := outputs[k] == secretSentinel
+		res[k] = OutputValue{
+			Value:  v,
+			Secret: isSecret,
+		}
+	}
+
+	return res, nil
+}
+
+// History returns a list summarizing all previous and current results from Stack lifecycle operations
+// (up/preview/refresh/destroy).
+func (s *Stack) History(ctx context.Context) ([]UpdateSummary, error) {
+	err := s.Workspace().SelectStack(ctx, s.Name())
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get stack history")
+	}
+
+	stdout, stderr, errCode, err := s.runPulumiCmdSync(ctx, "history", "--json", "--show-secrets")
+	if err != nil {
+		return nil, newAutoError(errors.Wrap(err, "failed to get stack history"), stdout, stderr, errCode)
+	}
+
+	var history []UpdateSummary
+	err = json.Unmarshal([]byte(stdout), &history)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to unmarshal history result")
+	}
+
+	return history, nil
+}
+
+// GetConfig returns the config value associated with the specified key.
+func (s *Stack) GetConfig(ctx context.Context, key string) (ConfigValue, error) {
+	return s.Workspace().GetConfig(ctx, s.Name(), key)
+}
+
+// GetAllConfig returns the full config map.
+func (s *Stack) GetAllConfig(ctx context.Context) (ConfigMap, error) {
+	return s.Workspace().GetAllConfig(ctx, s.Name())
+}
+
+// SetConfig sets the specified config KVP.
```

Comment:
I'm guessing that's what KVP is
```suggestion
// SetConfig sets the specified config kev-value pair.
```